examples/pytorch/graphsage_base/node_classification.py

```diff
@@ -0,0 +1,141 @@
+                h = self.dropout(h)
+        return h
+
+    def inference(self, g, device, batch_size):
+        """Conduct layer-wise inference to get all the node embeddings."""
+        feat = g.ndata['feat']
+        sampler = dgl.dataloading.MultiLayerFullNeighborSampler(1, prefetch_node_feats=['feat'])
+        dataloader = dgl.dataloading.DataLoader(
+                g, torch.arange(g.num_nodes()).to(g.device), sampler, device=device,
+                batch_size=batch_size, shuffle=False, drop_last=False,
+                num_workers=0)
+        buffer_device = 'cpu'
+
+        for l, layer in enumerate(self.layers):
+            y = torch.empty(
+                g.num_nodes(), self.hid_size if l != len(self.layers) - 1 else self.out_size,
+                device=buffer_device, pin_memory=True)
+            feat = feat.to(device)
+            for input_nodes, output_nodes, blocks in tqdm.tqdm(dataloader):
+                # use an explicitly contiguous slice
+                x = feat[input_nodes]
+                h = layer(blocks[0], x)
+                if l != len(self.layers) - 1:
+                    h = F.relu(h)
+                    h = self.dropout(h)
+                # by design, our output nodes are contiguous so we can take
+                # advantage of that here
+                y[output_nodes[0]:output_nodes[-1]+1] = h.to(buffer_device)
+            feat = y
+        return y
+    
+def evaluate(model, graph, dataloader):
+    model.eval()
+    ys = []
+    y_hats = []
+    for it, (input_nodes, output_nodes, blocks) in enumerate(dataloader):
+        with torch.no_grad():
+            x = blocks[0].srcdata['feat']
+            ys.append(blocks[-1].dstdata['label'])
+            y_hats.append(model(blocks, x))
+    return MF.accuracy(torch.cat(y_hats), torch.cat(ys))
+
+def layerwise_infer(args, device, graph, nid, model, batch_size):
+    model.eval()
+    nid = nid.to(device)
+    with torch.no_grad():
+        pred = model.inference(graph, device, batch_size)
+        pred = pred[nid].to(device)
+        label = graph.ndata['label'][nid].to(device)
+        return MF.accuracy(pred, label)
+
+def train(args, device, g, dataset, model):
+    # create sampler & dataloader
+    train_idx = dataset.train_idx.to(device)
+    val_idx = dataset.val_idx.to(device)
+    sampler = dgl.dataloading.NeighborSampler([15, 10, 5],
+                                              prefetch_node_feats=['feat'],
+                                              prefetch_labels=['label'])
+    use_uva = (args.mode == 'mixed')
+    train_dataloader = dgl.dataloading.DataLoader(g, train_idx, sampler, device=device,
+                                                  batch_size=1024, shuffle=True,
+                                                  drop_last=False, num_workers=0,
+                                                  use_uva=use_uva)
+
+    valid_dataloader = dgl.dataloading.DataLoader(g, val_idx, sampler, device=device,

```

Comment:
`val` above and `valid` here. It's better to be consistent. Personally I prefer `val` as `valid` seems to deem other dataloaders as "invalid" :)